File: common/src/main/java/com/lgw/android/common/http/base/BaseObserver.java
package com.lgw.android.common.http.base;


import android.widget.TextView;


import java.util.HashMap;
import java.util.Hashtable;
import java.util.List;

import io.reactivex.Observer;
import io.reactivex.annotations.NonNull;

import static com.lgw.android.common.http.utils.RxExceptionUtilKt.exceptionHandler;

/**
 * Created by lgw on 2020/11/10
 */
public abstract  class BaseObserver<T> implements Observer<BaseResponse<T>> {


    @Override
    public void onNext(@NonNull BaseResponse<T> baseResponse) {
       if(baseResponse.getCode()==200){
           onSuccess(baseResponse.getContent());
       }else{
           onFailure(null,baseResponse.getErrMsg());
       }


    }

    @Override
    public void onError(@NonNull Throwable e) {
        onFailure(e, exceptionHandler(e));
    }

    public abstract void onSuccess(T t);

    public abstract void onFailure(Throwable e,String errorMsg);







}

File: jetpack/src/main/java/com/lgw/android/jetpack/utils/BaseObserver.kt
package com.lgw.android.jetpack.utils

import com.lgw.android.jetpack.bean.BaseResponse
import io.reactivex.Observer

/**
 *Created by lgw on 1/28/21
 */
abstract class BaseObserver<T> : Observer<BaseResponse<T>> {

    companion object {
        private const val TAG = "BaseObserver"
        private const val SUCCESS_CODE = 200
    }

    override fun onNext(response: BaseResponse<T>) {
        //对基础数据进行统一处理
        if(response.res_code== SUCCESS_CODE){
           onSuccess(response.demo)
        }else{
            onFailure(null,response.err_msg)
        }
    }


    override fun onError(e: Throwable) {

        onFailure(e,RxExceptionUtil.exceptionHandler(e))

    }




    abstract  fun onSuccess(result:T)

    abstract fun  onFailure(e: Throwable?, errorMsg:String)





}